hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -1333,14 +1334,33 @@ public static boolean scmInit(OzoneConfiguration conf,
       // Initialize security if security is enabled later.
       initializeSecurityIfNeeded(conf, scmStorageConfig, selfHostName, true);
 
-      if (SCMHAUtils.isSCMHAEnabled(conf) && !isSCMHAEnabled) {
+      if (conf.getBoolean(ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY,
+          ScmConfigKeys.OZONE_SCM_HA_ENABLE_DEFAULT) && !isSCMHAEnabled) {
         SCMRatisServerImpl.initialize(scmStorageConfig.getClusterID(),
             scmStorageConfig.getScmId(), haDetails.getLocalNodeDetails(),
             conf);
         scmStorageConfig.setSCMHAFlag(true);
         scmStorageConfig.setPrimaryScmNodeId(scmStorageConfig.getScmId());
         scmStorageConfig.forceInitialize();
-        LOG.debug("Enabled SCM HA");
+
+        /*
+         * Since Ratis is initialized on an existing cluster, we have to
+         * trigger Ratis snapshot so that this SCM can send the latest scm.db
+         * to the bootstrapping SCMs later.
+         */
+
+        try {
+          DefaultConfigManager.forceUpdateConfigValue(
+              ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY, true);
+          StorageContainerManager scm = createSCM(conf);
```

Comment:
Metrics system complains a bit here.  It doesn't happen in integration test, because `MiniOzoneCluster` needs to allow duplicate metrics anyway (due to running multiple components in the same JVM).

```
org.apache.hadoop.metrics2.MetricsException: Hadoop:service=LayoutVersionManager,name=HDDSLayoutVersionManager already exists!
	at org.apache.hadoop.metrics2.lib.DefaultMetricsSystem.newObjectName(DefaultMetricsSystem.java:135)
	at org.apache.hadoop.metrics2.lib.DefaultMetricsSystem.newMBeanName(DefaultMetricsSystem.java:110)
	at org.apache.hadoop.metrics2.util.MBeans.getMBeanName(MBeans.java:165)
	at org.apache.hadoop.metrics2.util.MBeans.register(MBeans.java:97)
	at org.apache.hadoop.metrics2.util.MBeans.register(MBeans.java:73)
	at org.apache.hadoop.ozone.upgrade.AbstractLayoutVersionManager.init(AbstractLayoutVersionManager.java:88)
	at org.apache.hadoop.hdds.upgrade.HDDSLayoutVersionManager.<init>(HDDSLayoutVersionManager.java:54)
	at org.apache.hadoop.hdds.scm.server.StorageContainerManager.initializeSystemManagers(StorageContainerManager.java:675)
	at org.apache.hadoop.hdds.scm.server.StorageContainerManager.<init>(StorageContainerManager.java:411)
	at org.apache.hadoop.hdds.scm.server.StorageContainerManager.createSCM(StorageContainerManager.java:612)
	at org.apache.hadoop.hdds.scm.server.StorageContainerManager.createSCM(StorageContainerManager.java:624)
	at org.apache.hadoop.hdds.scm.server.StorageContainerManager.scmInit(StorageContainerManager.java:1355)
	at org.apache.hadoop.hdds.scm.server.StorageContainerManagerStarter$SCMStarterHelper.init(StorageContainerManagerStarter.java:186)
	at org.apache.hadoop.hdds.scm.server.StorageContainerManagerStarter.initScm(StorageContainerManagerStarter.java:116)
```

Reply:
Yes, this needs to be fixed. I'm planning on introducing separate classloaders for different components in `MiniOzoneCluster`. Hopefully, that will solve this problem.

Reply:
@nandakumar131 the exception I showed is from acceptance test, which runs each component in separate JVM, not from `MiniOzoneCluster`

Reply:
Ah! My bad. let me fix the metric issue here.

Reply:
Fixed the metric issue.